doc/modules/ROOT/pages/tutorials/visualize-with-pyvis.adoc

```diff
@@ -0,0 +1,201 @@
+are citations.
+
+We will then sample a smaller representative subgraph from it that is
+more suitable for visualization.
+
+[source, python, role=no-test]
+----
+G = gds.graph.load_cora()
+----
+
+Let’s make sure we constructed the correct graph.
+
+[source, python, role=no-test]
+----
+print(f"Metadata for our loaded Cora graph `G`: {G}")
+print(f"Node labels present in `G`: {G.node_labels()}")
+----
+
+It’s looks correct! Now let’s go ahead and sample the graph.
+
+We use the random walk with restarts sampling algorithm to get a smaller
+graph that structurally represents the full graph. In this example we
+will use the algorithm’s default parameters, but check out
+https://neo4j.com/docs/graph-data-science/current/management-ops/graph-creation/sampling/rwr/[the
+algorithm’s docs] to see how you can for example specify the size of the
+subgraph, and choose which start node around which the subgraph will be
+sampled.
+
+[source, python, role=no-test]
+----
+G_sample, _ = gds.alpha.graph.sample.rwr("cora_sample", G, randomSeed=42, concurrency=1)
+----
+
+We should have somewhere around 0.15 * 2708 ~ 406 nodes in our sample.
+And let’s see how many relationships we got.
+
+[source, python, role=no-test]
+----
+print(f"Number of nodes in our sample: {G_sample.node_count()}")
+print(f"Number of relationships in our sample: {G_sample.relationship_count()}")
+----
+
+Let’s also compute
+https://neo4j.com/docs/graph-data-science/current/algorithms/page-rank/[PageRank]
+on our sample graph, in order to get an importance score that we call
+``rank'' for each node. It will be interesting for context when we
+visualize the graph.
+
+[source, python, role=no-test]
+----
+gds.pageRank.mutate(G_sample, mutateProperty="rank")
+----
+
+== Exporting the sampled Cora graph
+
+We can now export the topology and node properties of our sampled graph
+that we want to visualize.
+
+Let’s start by fetching the relationships.
+
+[source, python, role=no-test]
+----
+sample_topology_df = gds.beta.graph.relationships.stream(G_sample)
+display(sample_topology_df)
+----
+
+We get the right amount of rows, one for each expected relationship. So
+that looks good.
+
+Next we should fetch the node properties we are interested in. Each node
+will have a ``subject'' property which will be an integer 0,…,6 that
+indicates which of seven academic subjects the paper represented by the
+nodes belong to. We will also fetch the PageRank property ``rank'' that
+we computed above.
+
+[source, python, role=no-test]
+----
+sample_node_properties_df = gds.graph.nodeProperties.stream(
+    G_sample,
+    ["subject", "rank"],
+    separate_property_columns=True,
+)
+display(sample_node_properties_df)
+----
+
+Now that we have all the data we want to visualize, we can create a
+network with PyVis. We color each node according to its ``subject'', and
+size it according to its ``rank''.
+
+[source, python, role=no-test]
+----
+net = Network(notebook = True,
+cdn_resources="remote",
```

Comment:
I though `ruff` would fix this ugly indentation?

Reply:
likely wrong pandoc version installed locally.